shell/platform/android/io/flutter/embedding/engine/systemchannels/TextInputChannel.java

```diff
@@ -257,7 +314,14 @@ public static Configuration fromJson(@NonNull JSONObject json)
       if (inputActionName == null) {
         throw new JSONException("Configuration JSON missing 'inputAction' property.");
       }
-
+      Configuration[] fields = null;
+      if (!json.isNull("fields")) {
+        final JSONArray fields = json.getJSONArray("fields");
+        fields = new Configuration[fields.length()];
```

Comment:
There are two declarations of `fields` above, maybe some code got mixed up or missed being deleted.